### ISSUE TITLE
"Audio source" settings - Clearer names or Add documentation for that

### ISSUE THREAD

**Issue by inson1** [FossifyOrg/Voice-Recorder#16]
**Checklist**
 - [x] I made sure that there are no existing issues - open or closed - to which I could contribute my information.
 - [x]  I have read the FAQ and my problem isn't listed.
 - [x]  I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
 - [x]  This issue contains only one feature request.
 - [x]  I have read and understood the [contribution guidelines.](https://github.com/FossifyOrg/General-Discussion?tab=readme-ov-file#faq)

**Is your feature request related to a problem? Please describe.**
I'm quite confused by the options for "Audio source" in the app settings.
For example, what does "Android default" mean, and why isn't it the default option?

Could this setting be made clearer, perhaps with information about what each option means and how one would choose an audio source in the Readme?

**Describe the solution you'd like**
title - first would  be better

**Describe alternatives you've considered**
title

**Comment by inson1** [comment]
https://github.com/SimpleMobileTools/Simple-Voice-Recorder/issues/244

**Comment by Aga-C** [comment]
In Readme nearly nobody will find it. I think the best would be a short description under each radio button. Alternatively, a new entry in FAQ (in the app).

**Comment by tanyabouman** [comment]
I think the audio sources in the app correspond to some of the [MediaRecorder.AudioSource](https://developer.android.com/reference/android/media/MediaRecorder.AudioSource) options from Android.

**Comment by smutax** [comment]
Hi there. Would choosing any of the aforementioned audio sources allow recording of two-way voice or Voip communication? Like "call recorder". Thanks.

**Comment by tanyabouman** [comment]
> Hi there. Would choosing any of the aforementioned audio sources allow recording of two-way voice or Voip communication? Like "call recorder". Thanks.

I don't think you will be able to record calls because VOICE_CALL is an audio source option, but there are some options from the Android API documentation:

- [VOICE_COMMUNICATION](https://developer.android.com/reference/android/media/MediaRecorder.AudioSource#VOICE_COMMUNICATION) Microphone audio source tuned for voice communications such as VoIP. It will for instance take advantage of echo cancellation or automatic gain control if available.

- [VOICE_CALL](https://developer.android.com/reference/android/media/MediaRecorder.AudioSource#VOICE_CALL) Voice call uplink + downlink audio source
Capturing from VOICE_CALL source requires the Manifest.permission.CAPTURE_AUDIO_OUTPUT permission. This permission is reserved for use by system components and is not available to third-party applications.

**Comment by smutax** [comment]
> > Hi there. Would choosing any of the aforementioned audio sources allow recording of two-way voice or Voip communication? Like "call recorder". Thanks.
> 
> I don't think you will be able to record calls because VOICE_CALL is an audio source option, but there are some options from the Android API documentation:
> 
> * [VOICE_COMMUNICATION](https://developer.android.com/reference/android/media/MediaRecorder.AudioSource#VOICE_COMMUNICATION) Microphone audio source tuned for voice communications such as VoIP. It will for instance take advantage of echo cancellation or automatic gain control if available.
> * [VOICE_CALL](https://developer.android.com/reference/android/media/MediaRecorder.AudioSource#VOICE_CALL) Voice call uplink + downlink audio source
>   Capturing from VOICE_CALL source requires the Manifest.permission.CAPTURE_AUDIO_OUTPUT permission. This permission is reserved for use by system components and is not available to third-party applications.

Thanks for clarifying. I don't suppose you know, or if it's at all relevant. What Android API feature music recognition applications such as Shazam take advantage of, as they allow recognition of sound(music) internally , from another application, hence my assumption that they can capture audio output? Sorry if above is confusing.

**Comment by tanyabouman** [comment]
> Thanks for clarifying. I don't suppose you know, or if it's at all relevant. What Android API feature music recognition applications such as Shazam take advantage of, as they allow recognition of sound(music) internally , from another application, hence my assumption that they can capture audio output? Sorry if above is confusing.

I don't know anything about the internals of Shazam, but I'd guess that [Capture audio playback](https://developer.android.com/media/platform/av-capture#capture_audio_playback) is probably similar.

**Comment by miscounted** [comment]
Frequency response graphs will make it easier to decide which application it would suit the best.

As all android devices are different, onr way to know for sure is to capture a known calibration noise (like brown noise) with all of the presets. It would be convenient to have an in-app function that does that, even if the user would have to provide a brown noise source first.

**Comment by naveensingh** [comment]
This preference should be removed or repurposed to show options like:

- Auto-detect (external or phone mic)
- Always use phone mic

**Comment by naveensingh** [comment]
> Reverting to [1.3.3](https://github.com/FossifyOrg/Voice-Recorder/releases/tag/1.3.3) and setting the audio source to "Android default" fixes the issue for me.

Thanks for reporting. I'll switch it to "Android default" in the next version.

**Comment by Funkyflosh** [comment]
> Is anyone else facing any issues due to this removal?

Hi Naveen, first: I really  appreciate your great work!
My problem is, that my recordings are now very much less loud than they used to be.
Also using a Pixel mobile is a true PITA regarding audio (calls, recordings, etc.). Your app was the only one that gave me the option to choose an audio input and thus still use voice recordings. 
I'd be very grateful if you could bring back the feature in the next version.
Regards
Florian

**Comment by naveensingh** [comment]
Added the option back and renamed it to **Microphone mode.** Available in [v1.4.0](https://github.com/FossifyOrg/Voice-Recorder/releases/tag/1.4.0).
